### PR TITLE
Hive 0.11 compatibility

### PR DESCRIPTION
**Pull request by alupher**
Initial commit of support for Hive 0.11. This compiles and runs as expected, passing most regression tests, though more thorough testing is still in progress. Does not yet include support for new Hive 0.11 operators (e.g. PTFOperator) or new stats features.

### CODE REVIEW

File: src/main/scala/shark/parse/SharkExplainSemanticAnalyzer.scala [hunk 1]
```diff
@@ -62,7 +62,7 @@ class SharkExplainSemanticAnalyzer(conf: HiveConf) extends ExplainSemanticAnalyz
     }
 
     val task = TaskFactory.get(
-      new SharkExplainWork(ctx.getResFile().toString(), tasks, childNode.toStringTree(), extended),
+      new SharkExplainWork(ctx.getResFile().toString(), tasks, childNode.toStringTree(), sem.getInputs(), extended),
```
Comment (rxin) [inline src/main/scala/shark/parse/SharkExplainSemanticAnalyzer.scala]: wrap this at 100 col

File: src/test/java/shark/SharkQTestUtil.java [hunk 2]
```diff
@@ -89,6 +93,7 @@ public void cliInit(String tname, boolean recreate) throws Exception {
 
   @Override
   public int executeClient(String tname) {
+    LOG.info("Begin query: " + tname);
```
Comment (rxin) [inline src/test/java/shark/SharkQTestUtil.java:96]: Change this to "Begin test: "

Comment (alupher) [inline src/test/java/shark/SharkQTestUtil.java:96]: Looks like Hive uses "Begin query" (see line 126 of https://github.com/apache/hive/blob/branch-0.11/ql/src/test/templates/TestCliDriver.vm) -- probably should keep the same for consistency.

**Comment by rxin** [comment]
Merging this in so we can update the hive 0.11 branch. Thanks, Antonio!

**Pull request merged**